File: robocode2-server/src/main/java/net/robocode2/events/BulletMissedEvent.java
package net.robocode2.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bullet has missed (reached the walls).
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BulletMissedEvent implements Event {

	/** Bullet that missed */
	Bullet bullet;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BulletMissedEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class BulletMissedEvent(
        val bullet: BulletState
) : Content(type = ContentType.BULLET_MISSED_EVENT.type)
